src/providers/script/base.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Notation:
+ *   - A "word" is a 32-bit interger
+ */
+import { ByteBuffer } from 'dojo-core/encoding';
+
+/**
+ * A script hash function
+ */
+export interface ScriptHash {
+	(data: ByteBuffer): ByteBuffer;
+	blockSize: number;
+}
+
+/**
+ * A general math function
+ */
+export interface MathFunction {
+	(...inputs: number[]): number
+}
+
+/**
+ * Add a list of words, with rollover
+ */
+export function addWords(...words: number[]): number {
+	const numWords = words.length;
+	let sum = words[0];
+	for (let i = 1; i < numWords; i++) {
+		const a = sum;
+		const b = words[i];
+		const low = (a & 0xFFFF) + (b & 0xFFFF);
+		const high = (a >> 16) + (b >> 16) + (low >> 16);
+		sum = (high << 16) | (low & 0xFFFF);
+	}
+	return sum;
```

Comment:
Hmmm...interesting point. The only time would be if you just called `addWords()` with no args, so that's probably fine.